MSAL/src/requests/MSALInteractiveRequest.m

```diff
@@ -270,7 +270,7 @@ - (MSIDTokenRequest *)tokenRequest
 {
     return [[MSIDAADAuthorizationCodeGrantRequest alloc] initWithEndpoint:[self tokenEndpoint]
                                                                  clientId:_parameters.clientId
-                                                             enrollmentId:nil
+                                                             enrollmentId:[_authority enrollmentIdForHomeAccountId:_parameters.account.homeAccountId.identifier]
```

Comment:
in case of migrating from ADAL to MSAL, enrollment ID might be saved under displayable identifier, so we should fallback to query by username from home account id.

Reply:
JZ mentioned that for MSAL-related enrollment ID work relying on homeAccountId should be enough - I assume he didn't think of data migration scenario?

Reply:
Yes, I think he didn't :) There might be an app that updates from ADAL to MSAL. MAM SDK used to only write displayable ID before. So, while we should always rely first on homeAccountId when it's present, we also need a fallback. There was already login in ADAL that first checked by home Account ID and fell back to legacy account Id.